src/systemctl/systemctl-logind.c

```diff
@@ -75,6 +80,32 @@ int logind_reboot(enum action a) {
         if (arg_dry_run)
                 return 0;
 
+        r = sd_bus_message_new_method_call(bus, &m, bus_login_mgr->destination, bus_login_mgr->path,
+                                           bus_login_mgr->interface, actions[a].method_with_flags);
+        if (r < 0)
+                return r;
+
+        SET_FLAG(flags, SD_LOGIND_ROOT_CHECK_INHIBITORS, arg_check_inhibitors);
+
+        r = sd_bus_message_append(m, "t", flags);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_set_allow_interactive_authorization(m, arg_ask_password);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_call(bus, m, 0, &error, NULL);
+        if (r < 0) {
+                if (!sd_bus_error_has_name(&error, SD_BUS_ERROR_UNKNOWN_METHOD))
+                        return log_error_errno(r, "Failed to %s via logind: %s", actions[a].description, bus_error_message(&error, r));
+        } else
+                return 0;
```

Comment:
hmm, i'd rewrite this as:

```c
if (r >= 0)
        return 0;
if (!sd_bus_error_has_name(&error, SD_BUS_ERROR_UNKNOWN_METHOD))
        return log_error_errno(r, …);
```

i.e. keep the indentation levels short, and prefer branches that exit quickly and let the main code flow continue on the top level indentation